apps/worker/src/request_processor/dsnp.announcement.processor.ts

```diff
@@ -0,0 +1,338 @@
+          await this.reactionQueue.add(`Reaction Job - ${data.id}`, reaction, { jobId: data.id, removeOnFail: false, removeOnComplete: 2000 });
+          break;
+        }
+        case AnnouncementTypeDto.UPDATE: {
+          const updateDto = data.content as UpdateDto;
+          const updateAnnoucementType: AnnouncementType =
+            updateDto.targetAnnouncementType === ModifiableAnnouncementTypeDto.BROADCAST ? AnnouncementType.Broadcast : AnnouncementType.Reply;
+          const update = await this.processUpdate(updateDto, updateAnnoucementType, updateDto.targetContentHash ?? '', data.dsnpUserId);
+          await this.updateQueue.add(`Update Job - ${data.id}`, update, { jobId: data.id, removeOnFail: false, removeOnComplete: 2000 });
+          break;
+        }
+        case AnnouncementTypeDto.PROFILE: {
+          const profile = await this.processProfile(data.content as ProfileDto, data.dsnpUserId);
+          await this.profileQueue.add(`Profile Job - ${data.id}`, profile, { jobId: data.id, removeOnFail: false, removeOnComplete: 2000 });
+          break;
+        }
+        case AnnouncementTypeDto.TOMBSTONE: {
+          const tombStoneDto = data.content as TombstoneDto;
+          let targetAnnouncementType: AnnouncementType;
+          switch (tombStoneDto.targetAnnouncementType) {
+            case ModifiableAnnouncementTypeDto.BROADCAST: {
+              targetAnnouncementType = AnnouncementType.Broadcast;
+              break;
+            }
+            case ModifiableAnnouncementTypeDto.REPLY: {
+              targetAnnouncementType = AnnouncementType.Reply;
+              break;
+            }
+            default:
+              throw new Error(`Unsupported target announcement type ${typeof tombStoneDto.targetAnnouncementType}`);
+          }
+          const announcementType: AnnouncementType = targetAnnouncementType;
+          const tombstone = createTombstone(data.dsnpUserId, announcementType, tombStoneDto.targetContentHash ?? '');
+          await this.tombstoneQueue.add(`Tombstone Job - ${data.id}`, tombstone, { jobId: data.id, removeOnFail: false, removeOnComplete: 2000 });
+          break;
+        }
+        default:
+          throw new Error(`Unsupported announcement type ${typeof data.announcementType}`);
+      }
+    } catch (e) {
+      this.logger.error(`Error processing announcement ${data.announcementType} for ${data.dsnpUserId}: ${e}`);
+      throw e;
+    }
+  }
+
+  public async prepareNote(noteContent?: any): Promise<[string, string, string]> {
+    this.logger.debug(`Preparing note`);
+    const tags: ActivityContentTag[] = [];
+    if (noteContent?.content.tag) {
+      noteContent.content.tag.forEach((tag) => {
+        switch (tag.type) {
+          case TagTypeDto.Hashtag:
+            tags.push({ name: tag.name || '' });
+            break;
+          case TagTypeDto.Mention:
+            tags.push({
+              name: tag.name || '',
```

Comment:
this is not accurate, `name` is not a required field for mentions so if it is not set we should not set an empty field for it. there might be different meanings for an empty string and not set one.